### PR TITLE
Veoss 599 new mode

### PR DESCRIPTION
**Pull request by oliwiaKot**

### CODE REVIEW

File: app/pages/corrections/RemoveCountryCorrectionPage.scala [hunk 1]
```diff
@@ -45,4 +45,13 @@ case class RemoveCountryCorrectionPage(periodIndex: Index) extends QuestionPage[
         case _ => controllers.corrections.routes.CorrectPreviousReturnController.onPageLoad(CheckMode, answers.period)
       }
     }
+
+  override def navigateInCheckThirdLoopMode(answers: UserAnswers): Call =
+    answers.get(DeriveNumberOfCorrections(periodIndex)) match {
+      case Some(n) if n > 0 => controllers.corrections.routes.VatCorrectionsListController.onPageLoad(CheckThirdLoopMode, answers.period, periodIndex)
```
Comment (darbyo) [inline app/pages/corrections/RemoveCountryCorrectionPage.scala:51]: Can we rename n and x in here to more useful names

File: app/pages/corrections/VatCorrectionsListPage.scala [hunk 2]
```diff
@@ -33,7 +33,12 @@ case class VatCorrectionsListPage(periodIndex: Index) extends Page {
         case None => routes.JourneyRecoveryController.onPageLoad()
       }
     } else {
-      correctionRoutes.VatPeriodCorrectionsListController.onPageLoad(mode, answers.period)
+      if(mode == CheckThirdLoopMode){
+        correctionRoutes.VatPeriodCorrectionsListController.onPageLoad(NormalMode, answers.period)
+      }else{
```
Comment (darbyo) [inline app/pages/corrections/VatCorrectionsListPage.scala:38]: tiny one here but if making other changes can we add spaces before and after else 👍🏻

Comment (darbyo) [reply app/pages/corrections/VatCorrectionsListPage.scala:38]: theres quite a few places this applies, I wont comment them all

File: conf/app.routes [hunk 3]
```diff
@@ -148,6 +148,8 @@ GET         /:period/correctionCountry/:periodIndex/:countryIndex
 POST        /:period/correctionCountry/:periodIndex/:countryIndex                                            controllers.corrections.CorrectionCountryController.onSubmit(mode: Mode = NormalMode, period: Period, periodIndex: Index, countryIndex: Index)
 GET         /:period/changeCorrectionCountry/:periodIndex/:countryIndex                                      controllers.corrections.CorrectionCountryController.onPageLoad(mode: Mode = CheckMode, period: Period, periodIndex: Index, countryIndex: Index)
 POST        /:period/changeCorrectionCountry/:periodIndex/:countryIndex                                      controllers.corrections.CorrectionCountryController.onSubmit(mode: Mode = CheckMode, period: Period, periodIndex: Index, countryIndex: Index)
+GET         /:period/third-changeCorrectionCountry/:periodIndex/:countryIndex                                      controllers.corrections.CorrectionCountryController.onPageLoad(mode: Mode = CheckThirdLoopMode, period: Period, periodIndex: Index, countryIndex: Index)
```
Comment (darbyo) [inline conf/app.routes:151]: have we discussed urls with Joe? If not and dont want to do it on this PR can we make sure to put it in a ticket so we dont forget to do it

File: test/controllers/corrections/VatPeriodCorrectionsListControllerSpec.scala [hunk 4]
```diff
@@ -87,6 +88,24 @@ class VatPeriodCorrectionsListControllerSpec extends SpecBase with MockitoSugar
         Period(2022, Q1)
       )
 
+      val allPeriodsModel = Seq(
+        ListItem(
+          name = "1 July to 30 September 2021",
+          changeUrl = "/pay-vat-on-goods-sold-to-eu/northern-ireland-returns-payments" + controllers.corrections.routes.VatCorrectionsListController.onPageLoad(CheckThirdLoopMode, period, index).url,
+          removeUrl = "/pay-vat-on-goods-sold-to-eu/northern-ireland-returns-payments" + controllers.corrections.routes.RemovePeriodCorrectionController.onPageLoad(NormalMode, period, index).url
```
Comment (darbyo) [inline test/controllers/corrections/VatPeriodCorrectionsListControllerSpec.scala:95]: we could hard code these urls to make it more readable, again something to consider but up to yous

**Review by darbyo — CHANGES_REQUESTED** [review]

File: app/pages/corrections/RemoveCountryCorrectionPage.scala (hunk 1 above)
Comment (oliwiaKot) [reply app/pages/corrections/RemoveCountryCorrectionPage.scala]: done

File: app/pages/corrections/VatCorrectionsListPage.scala (hunk 2 above)
Comment (oliwiaKot) [reply app/pages/corrections/VatCorrectionsListPage.scala]: done

File: conf/app.routes (hunk 3 above)
Comment (oliwiaKot) [reply conf/app.routes:151]: we haven't, we created a ticket for that and put it in weekly priorities VEOSS-662

File: test/controllers/corrections/VatPeriodCorrectionsListControllerSpec.scala (hunk 4 above)
Comment (oliwiaKot) [reply test/controllers/corrections/VatPeriodCorrectionsListControllerSpec.scala]: done

**Review by darbyo — APPROVED** [review]

**Pull request merged**